### PR TITLE
refactor(ampd): switch from horizon to rpc client for Stellar

### PR DESCRIPTION
**Pull request by haiyizxx**
## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

File: ampd/src/stellar/rpc_client.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    ) -> error_stack::Result<HashMap<String, TxResponse>, Error> {
+        let tx_hashes: Vec<_> = tx_hashes
+            .into_iter()
+            .map(|tx_hash| Hash::from_str(tx_hash.as_str()).change_context(Error::TxHash))
+            .collect::<std::result::Result<Vec<_>, _>>()?;
+
+        let responses = join_all(
+            tx_hashes
+                .iter()
+                .map(|tx_hash| self.0.get_transaction(tx_hash)),
```
Comment (milapsheth) [inline ampd/src/stellar/rpc_client.rs:86]: Looks like we don't support doing a batched JSON RPC query in ampd? Seems to be the same as EVM. Is there a task to add this later?

Comment (milapsheth) [inline ampd/src/stellar/rpc_client.rs:86]: I imagine that `0x` prefix shouldn't be included for this rpc query as well?

**Review by milapsheth — APPROVED** [review]

File: ampd/src/stellar/rpc_client.rs (hunk 1 above)
Comment (haiyizxx) [reply ampd/src/stellar/rpc_client.rs:86]: The crate doesn’t support batch yet; we can update once they add that support.

Comment (haiyizxx) [reply ampd/src/stellar/rpc_client.rs:86]: yes, same as before. the handler trim the prefix

**Pull request merged**